pkg/controller/config.go

```diff
@@ -0,0 +1,147 @@
+package controller
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/featureflags"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DomainDefault                      = "https://cloud.mongodb.com/"
+	ObjectDeletionProtectionDefault    = true
+	SubObjectDeletionProtectionDefault = true
+	operatorPodNameDefault             = "mongodb-atlas-operator"
+	operatorNamespaceDefault           = "default"
+
+	objectDeletionProtectionFlag      = "object-deletion-protection"
+	subobjectDeletionProtectionFlag   = "subobject-deletion-protection"
+	objectDeletionProtectionEnvVar    = "OBJECT_DELETION_PROTECTION"
+	subobjectDeletionProtectionEnvVar = "SUBOBJECT_DELETION_PROTECTION"
+)
+
+type Config struct {
+	Domain             string
+	APISecret          client.ObjectKey
+	DeletionProtection DeletionProtection
+	FeatureFlags       *featureflags.FeatureFlags
+}
+
+type DeletionProtection struct {
+	Object    bool
+	SubObject bool
+}
+
+func (c *Config) ParseFlagsFromEnv(fs *flag.FlagSet) {
+	if c == nil {
+		return
+	}
+
+	parseDeletionProtection(c, fs)
+	c.FeatureFlags = featureflags.NewFeatureFlags(os.Environ)
+}
```

Comment:
IMHO as said before CLI stuff belongs on the main code. Internal code should be agnostic on how the config was put together, which should also make it more natural for internal code calling this or other consumers of the code.